src/main/java/com/revature/responsecollector/JMeterResponseCollector.java

```diff
@@ -0,0 +1,178 @@
+    
+    private static final long serialVersionUID = 1L;
+    private List<Long> responseTimes = new ArrayList<>();
+    private long responseMax = 0;
+    
+    private long firstSampleStartTime = 0;
+    private long sampleStartTime;
+    
+    // count of number of nXX status codes
+    private int num1XX;
+    private int num2XX;
+    private int num3XX;
+    private int num4XX;
+    private int num5XX;
+    
+    public JMeterResponseCollector(Summariser summer) {
+        super(summer);
+    }
+
+    @Override
+    public void sampleOccurred(SampleEvent e) {
+        super.sampleOccurred(e);
+        SampleResult r = e.getResult();
+        if (firstSampleStartTime == 0) {
+            firstSampleStartTime = r.getStartTime();
+        }
+        sampleStartTime = r.getStartTime();
+        long responseTime = r.getEndTime() - r.getStartTime();
+        responseTimes.add(responseTime);
+        if (responseTime > responseMax) {
+            responseMax = responseTime;
+        }
+        
+        char respType = r.getResponseCode().charAt(0);
+        
+        switch (respType) {
+        case '1':
+            num1XX++;
+            break;
+        case '2':
+            num2XX++;
+            break;
+        case '3':
+            num3XX++;
+            break;
+        case '4':
+            num4XX++;
+            break;
+        case '5':
+            num5XX++;
+            break;
+        default:
+            // TODO log error 
+        }
+
+    }
+    
+    public double getsuccessFailPercentage() {
+        double ratio = 0;
+        if (num2XX + num4XX + num5XX != 0) {
+            ratio = ((double) num2XX) / (num2XX + num4XX + num5XX);
+        }
+        return ratio;
+    }
+    
+    public long getResponseAvg() {
+        long length = responseTimes.size();
+        long sum = 0;
+        for (long lat : responseTimes) {
+            sum += lat;
+        }
+        return sum / length;
+    }
+    
+    public long getResponse50Percentile() {
+        Collections.sort(responseTimes);
+
+        int middle = (int) Math.round(responseTimes.size() * 0.5);
+        return responseTimes.get(middle - 1);
+    }
+    
+    public long getResponse25Percentile() {
+        Collections.sort(responseTimes);
+
+        int split = (int) Math.round(responseTimes.size() * 0.25);
+        return responseTimes.get(split - 1);
+    }
+    
+    public long getResponse75Percentile() {
+        Collections.sort(responseTimes);
+        int split = (int) Math.round(responseTimes.size() * 0.75);
+        return responseTimes.get(split - 1);
+    }
+    
+    public double getReqPerSec() {
+        long duration = sampleStartTime - firstSampleStartTime;
+        double reqPerSec = 0;
+        if (duration != 0) {
+            reqPerSec = 1000 * ((double) responseTimes.size()) / duration;
+        }
+        return reqPerSec;
+    }
+
+    public List<Long> getResponseTimes() {
+        return responseTimes;
+    }
+
+
```

Comment:
Extra blank lines can likely be removed

Reply:
Miscellaneous getters/setters could be generated with Lombok if truly needed.

Reply:
We've decided to just utilize regular getters and setters instead of using Lombok.